processing/src/main/java/org/apache/druid/segment/data/GenericIndexed.java

```diff
@@ -508,47 +508,49 @@ public void close()
 
     int count = 0;
 
-    HeapByteBufferWriteOutBytes headerOut = new HeapByteBufferWriteOutBytes();
-    HeapByteBufferWriteOutBytes valuesOut = new HeapByteBufferWriteOutBytes();
-    try {
-      T prevVal = null;
-      do {
-        count++;
-        T next = objects.next();
-        if (allowReverseLookup && prevVal != null && !(strategy.compare(prevVal, next) < 0)) {
-          allowReverseLookup = false;
+    try (HeapByteBufferWriteOutBytes valuesOut = new HeapByteBufferWriteOutBytes()) {
```

Comment:
could you collapse this into a single try?

```
  try (
        HeapByteBufferWriteOutBytes valuesOut = new HeapByteBufferWriteOutBytes()
        HeapByteBufferWriteOutBytes headerOut = new HeapByteBufferWriteOutBytes();
    ) {
...
```

Reply:
Also, I'm not sure this actually needs closed either, from `WriteOutBytes` javadoc:

```
 * WriteOutBytes is a resource that is managed by {@link SegmentWriteOutMedium}, so it's own {@link #close()} method
 * does nothing. However WriteOutBytes should appear closed, i. e. {@link #isOpen()} returns false, after the parental
 * SegmentWriteOutMedium is closed.
```
Maybe we just ignore this one and leave it as is?